realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/SyncSessionTests.java

```diff
@@ -455,4 +462,121 @@ public void run() {
         assertEquals(3, realm.where(StringOnly.class).count());
         realm.close();
     }
+
+    @Test
+    public void partialSync() throws InterruptedException {
```

Comment:
Move this to its own class? I suspect we will have to add a lot more tests for this down the line?